Speed up index loading from SQLite for SeqIO, SearchIO, etc

pkerpedjiev opened the pull request:
The current `SELECT COUNT(key) FROM offset_data;` call to obtain the number of sequences in the database is slow for substantial fasta files.

Because the index is read-only, this can be replaced with `SELECT MAX(_ROWID_) FROM offset_data;` for orders of magnitude better performance.

For example, on a fasta with 17M sequences, loading the index with the current `COUNT(key)` call takes about 4 seconds. With `MAX(_ROWID_)` it takes 2.64ms.

On a fasta file with 307M sequences, loading the index with `COUNT(key)` takes 1 minute and 20 seconds. With `MAX(_ROWID_)` it takes 10ms. 

<!--- Please read each of the following items and confirm by replacing
 !--the [ ] with a [X] --->

- [x] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [x] I have read the ``CONTRIBUTING.rst`` file, have run ``flake8`` locally, and
understand that AppVeyor and TravisCI will be used to confirm the Biopython unit
tests and style checks pass with these changes.

- [x] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

peterjc commented:
Discussion on eg https://stackoverflow.com/questions/8988915/sqlite-count-slow-on-big-tables suggests this is a known SQLite weakness, and common work around.

The only reason we're doing this check is for data consistency, so I think your change is fine - even given the possibility of other tools/scripts building or modifying our SQLite index format.

However, I do think the code change does need some explanation as a comment, e.g.

```python
# Using "SELECT MAX(key) FROM offset_data;" is too slow
```

@bow Do you agree?

@pkerpedjiev Do you want to put your name in ``NEWS.rst`` too?

pkerpedjiev commented:
@peterjc Thanks for the quick response! I added a comment with a link to the StackOverflow question and a short explanation and also added my name to NEWS.rst.

peterjc commented:
Merged, thank you 👍
